### DISCUSSION TITLE
using month(date) togheter with mean(value) for multiple years

### DISCUSSION THREAD

**Discussion by dataagile-ema** [vega/altair#2620] [Q&A] [answered]
Hi!

In this example:
alt.Chart(temps).mark_bar().encode(
    x='month(date):O',
    y='mean(temp):Q'
)
If there are multiple years in the series ‘date’, the chart will show a mean for all the occurrences of the same month (june for example) from the different years included in ‘date’.
Is there a recommended way to get around this? I want a mean per month but separating june 2020 from june 2021.

Best Regards

**Comment by mattijn** [comment, marked as answer]
Use `yearmonth()` instead of `month()`.

**Reply by dataagile-ema** [reply]
great, thanks!
